File: cpp-answers/118.杨辉三角.cpp
/*
 * @Author: your name
 * @Date: 2020-06-03 13:45:23
 * @LastEditTime: 2020-06-05 20:33:29
 * @LastEditors: Please set LastEditors
 * @Description: In User Settings Edit
 * @FilePath: \postwoman-mastere:\leetcode刷题\118.杨辉三角.cpp
 */
/*
 * @lc app=leetcode.cn id=118 lang=cpp
 *
 * [118] 杨辉三角
 */

// @lc code=start
class Solution
{
public:
    vector<vector<int>> generate(int numRows)
    {
        vector<vector<int>> result;
        vector<int> row;
        for (int i = 0; i < numRows; i++)
        {
            row.push_back(1);
            for (int j = i - 1; j > 0; j--)
            {
                row[j] = row[j - 1] + row[j];
            }
            result.push_back(row);
        }
        return result;
    }
};
// @lc code=end

File: rust-answers/118.杨辉三角.rs
/*
 * @lc app=leetcode.cn id=118 lang=rust
 *
 * [118] 杨辉三角
 */

// @lc code=start
impl Solution {
    pub fn generate(num_rows: i32) -> Vec<Vec<i32>> {
        let mut vec = vec![vec![1]];

        for n in 1..num_rows {
            let mut sub_vec = vec![1];
            let mut prev_row = (n - 1) as usize;
            let mut prev_row_len = vec[prev_row].len();

            for i in 0..prev_row_len {
                if i == 0 {
                    sub_vec.push(vec[prev_row][i]);
                } else {
                    sub_vec.push(vec[prev_row][i]);
                    sub_vec[i] += vec[prev_row][i];
                }
            }
            vec.push(sub_vec);
        }
        vec
    }
}
// @lc code=end

